packages/flutter/lib/src/foundation/_isolates_io.dart

```diff
@@ -83,18 +83,30 @@ class _IsolateConfiguration<Q, R> {
   FutureOr<R> apply() => callback(message);
 }
 
-Future<void> _spawn<Q, R>(_IsolateConfiguration<Q, FutureOr<R>> configuration) async {
+/// The spawn point MUST guarantee only one result event is sent through the
+/// [SendPort.send] be it directly or indirectly i.e. [Isolate.exit].
+///
+/// In case an [Error] or [Exception] are thrown AFTER the data
+/// is sent, they will NOT be handled or reported by the main [Isolate] because
+/// it stops listening after the first event is received.
+///
+/// Also the awaited result from the [configuration.callback] has to be wrapped
+/// in a [List].
+Future<void> _spawn<Q, R>(_IsolateConfiguration<Q, R> configuration) async {
   final R result = await Timeline.timeSync(
     configuration.debugLabel,
-    () async {
-      final FutureOr<R> applicationResult = await configuration.apply();
-      return await applicationResult;
-    },
+    configuration.apply,
```

Comment:
I also saw timing in _spawn can be simplified, and applied the fix as well.

Both solutions side by side in a contrived example:
```dart
import 'dart:async';
import 'dart:developer';

void main() async {
  Timeline.timeSync('test', compute);
  print('finished');

  unawaited(Timeline.timeSync<Future<int>>('test', () async {
    final FutureOr<int> applicationResult = await compute();
    return await applicationResult;
  }));
  print('finished');
}

int _count = 0;
FutureOr<int> compute() async {
  final i = fib(40);

  print('DID ${_count++}');

  return i;
}

int fib(int i) {
  if (i < 2) return i;

  return fib(i - 2) + fib(i - 1);
}
```